Archives/VolFile.cpp

```diff
@@ -329,36 +318,37 @@ namespace Archives
 
 	bool VolFile::PrepareHeader(CreateVolumeInfo &volInfo)
 	{
-		IndexEntry *tempPtr;
-
-		if (OpenAllInputFiles(volInfo) == false) {
-			return false;// Verify input files can be read
+		if (!OpenAllInputFiles(volInfo)) {
+			return false;
 		}
 
-		// Get files sizes and calculate length of string table
+		// Get file sizes and calculate length of string table
 		for (std::size_t i = 0; i < volInfo.fileCount(); i++)
 		{
-			// Store the indexing info into the index struct
-			volInfo.indexEntry[i].fileSize = GetFileSize(volInfo.fileHandle[i], 0);
-			volInfo.indexEntry[i].fileNameOffset = volInfo.stringTableLength;
-			volInfo.indexEntry[i].compressionType = CompressionType::Uncompressed;
-			// Calculate length of internal file names for string table
-			volInfo.fileNameLength[i] = volInfo.internalNames[i].size() + 1; //Accounts for null terminator in size
-			volInfo.stringTableLength += volInfo.fileNameLength[i];
+			IndexEntry indexEntry;
+
+			indexEntry.fileSize = GetFileSize(volInfo.fileHandles[i], 0);
+			indexEntry.fileNameOffset = volInfo.stringTableLength;
+			indexEntry.compressionType = CompressionType::Uncompressed;
+
+			volInfo.indexEntries.push_back(indexEntry);
+
+			// Add length of internal fileName plus null terminator to string table length.
+			volInfo.stringTableLength += volInfo.internalNames[i].size() + 1;
 		}
 
 		// Calculate size of index table
 		volInfo.indexTableLength = volInfo.fileCount() * sizeof(IndexEntry);
 		// Calculate the zero padded length of the string table and index table
 		volInfo.paddedStringTableLength = (volInfo.stringTableLength + 7) & ~3;
 		volInfo.paddedIndexTableLength = (volInfo.indexTableLength + 3) & ~3;
-		volInfo.indexEntry[0].dataBlockOffset = volInfo.paddedStringTableLength + volInfo.paddedIndexTableLength + 32;
+		volInfo.indexEntries[0].dataBlockOffset = volInfo.paddedStringTableLength + volInfo.paddedIndexTableLength + 32;
 		
 		// Calculate offsets to the files
 		for (std::size_t i = 1; i < volInfo.fileCount(); i++)
 		{
-			tempPtr = &volInfo.indexEntry[i - 1];
-			volInfo.indexEntry[i].dataBlockOffset = (tempPtr->dataBlockOffset + tempPtr->fileSize + 11) & ~3;
+			IndexEntry previousIndex = volInfo.indexEntries[i - 1];
```

Comment:
I think we should use a reference type here:
```
IndexEntry& previousIndex = volInfo.indexEntries[i - 1];
```
A non-reference type will call the copy constructor and create a duplicate of the object. That is less efficient. Using the reference type does improve the syntax, while retaining the efficiency of using a pointer.

Might even make it a `const` reference, since it's just being used to read.